SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -450,11 +463,14 @@ - (void)sendMessageToService:(TSOutgoingMessage *)message
         TSThread *thread = message.thread;
 
         if ([thread isKindOfClass:[TSGroupThread class]]) {
+            OWSAssert(message.intendedRecipientIds);
+
             TSGroupThread *gThread = (TSGroupThread *)thread;
 
             NSError *error;
-            NSArray<SignalRecipient *> *recipients =
-                [self getRecipients:gThread.groupModel.groupMemberIds error:&error];
+            NSArray<SignalRecipient *> *recipients = [self getRecipients:gThread.groupModel.groupMemberIds
```

Comment:
If we re-send a failed message, only send to the intersection of:

* The original intended recipients.
* The current members of the thread.